### ISSUE TITLE
Looping/wrapping/repeating tile layers

### ISSUE THREAD

**Issue by exelotl** [mapeditor/tiled#3456]
Hi! I'm working on a game with levels that repeat on both axes. I frequently use the Offset Map feature to wrap the level around so I can focus on different parts of it, but it would be a nice quality of life improvement to be able to visualise the looping level while editing.

This might make sense as a per-layer property that could be toggled for each axis. Even in games that don't feature looping levels it might be useful as something people could apply to background layers. I guess it would only be for tile and image layers but not for object layers?

Alternatively it might work as a checkbox in the View menu, similar to what Aseprite offers.
_(in this case the value appears to be remembered on a per-tab basis, it's not actually global, so I suppose that's really just Aseprite's version of a map property?)_
![Screenshot from 2022-08-19 12-46-20](https://user-images.githubusercontent.com/569607/185611821-27d334e2-e219-4603-94e7-a5e246b831ef.png)

**Comment by exelotl** [comment]
> I guess it would only be for tile and image layers but not for object layers?

I just realised this feature already exists for image layers! So this issue is then a request for the same feature to be added to Tile layers.

<img src="https://user-images.githubusercontent.com/569607/190501685-62fd1428-b60f-4f4b-a952-f8bd0c5c6904.png" width=300>

**Comment by bjorn** [comment]
> I just realised this feature already exists for image layers!

Yes, and since an image layer can refer to a map to use as its image, you can even use that repetition feature to create a map that will preview another map as repeated. Or, you could create another map with just the tile layer you want to repeat, and add that map as a repeating image layer.

That isn't particularly great in terms of usability though, since the repeating map or tile layer can't be directly edited in that case. So, this issue will remain open regarding that feature.

**Comment by CelticMinstrel** [comment]
I second this request!

I note that you can actually make an image layer that refers _the same same map_ that contains it, allowing you to visually see the repeating map as you edit it. However, it's very limited, as it never refreshes to reflect changes as you edit. You need to manually refresh it by editing the Image Source property on the image layer.

But, that does suggest that maybe this isn't that difficult to implement directly.
